### PR TITLE
fix: add llm `drop_params` parameter to LLMConfig

### PR DESCRIPTION
**Pull request by tobitege**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

Some LLM providers do not accept any or all LLM parameters when called, which causes an **UnsupportedParamsError** exception.
This new parameter prevents this exception (see below for more details).

**Give a brief summary of what the PR does, explaining any non-trivial design decisions**

In communicating with the LLM there is a `Stop` parameter sent with every CodeAct*Agent request.
However, not all providers of LLM's support any and/or specific parameters (see end of text).
For e.g. **Mistral** users, this parameter leads to an exception, which bubbles up via litellm.

This PR's introduction of the new LLM parameter `drop_params` (`llm` section in `config.toml`)
should instruct the provider to drop any unsupported parameter and not raise an exception.

_Tested this flag set via config.toml, but my providers do support Stop, so this needs to
be tested in addition by someone else with e.g. Mistral or Cohere etc._

Should fix these 2 issues:
https://github.com/OpenDevin/OpenDevin/issues/2421 _Unable to use cohere command LLM_
https://github.com/OpenDevin/OpenDevin/issues/2464 _Error when using Mistral codestral_

**Other references**

**Log example:**
```Py
File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 5863, in get_optional_params
_check_valid_arg(supported_params=supported_params)
File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 5339, in _check_valid_arg
raise UnsupportedParamsError(
litellm.utils.UnsupportedParamsError: mistral does not support parameters: {'stop': ['</execute_ipython>', '</execute_bash>', '</execute_browse>']}, for model=codestral-latest. To drop these, set litellm.drop_params=True or for proxy:

litellm_settings: drop_params: true
```

The following providers are allegedly supporting the `Stop` parameter:
* Anthropic
* OpenAI
* Azure OpenAI
* Cohere (unsure about this)
* Huggingface
* Openrouter
* AI21

### CODE REVIEW

**Comment by SmartManoj** [comment]
I think Litellm docs are outdated. Eg: Groq doesn't support 'llama2-70b-4096' now and [the docs is not updated](https://litellm.vercel.app/docs/providers/groq#:~:text=os.environ%5B%27GROQ_API_KEY%27%5D-,Sample%20Usage,-from%20litellm%20import).

---
Here the error hint indicates only that
`To drop these, set litellm.drop_params=True or for proxy: litellm_settings: drop_params: true`

But [the doc says](https://litellm.vercel.app/docs/completion/input#input-params-1:~:text=To%20drop%20the%20param%20instead%2C%20set%20litellm.drop_params%20%3D%20True%20or%20completion(..drop_params%3DTrue).) 
`To drop the param instead, set litellm.drop_params = True or completion(..drop_params=True).`

Raises a question about the function param. 

Searched for the param in the function. And seems there is no reference to it. 
https://github.com/BerriAI/litellm/blob/3a35a58859a145a4a568548316a1930340e7440a/litellm/main.py#L463


Also passing LITELLM_DROP_PARAMS is enough. https://github.com/BerriAI/litellm/blob/3a35a58859a145a4a568548316a1930340e7440a/litellm/__init__.py#L76

**Comment by tobitege** [comment]
> Also passing LITELLM_DROP_PARAMS is enough.
You mean for use as part of a docker command. Otherwise it'd need to be an env variable again, or?

The source of the exception, as raised in the issues, is in litellm's file `utils.py`:
https://github.com/BerriAI/litellm/blob/3a35a58859a145a4a568548316a1930340e7440a/litellm/utils.py#L2105

**Comment by SmartManoj** [comment]
> Searched for the param in the function. And seems there is no reference to it.

Still got the same error. No effects with this.

```python
import os
from litellm import completion

os.environ["AWS_ACCESS_KEY_ID"] = "1"
os.environ["AWS_SECRET_ACCESS_KEY"] = "1"
os.environ["AWS_REGION_NAME"] = "ap-southeast-1"

response = completion(
    model="bedrock/cohere.command-r-plus-v1:0",
    messages=[{"content": "Hello, how are you?", "role": "user"}],
    stop=["\n", "Human:", "AI:"],
    drop_params = True,
)
```

**Comment by SmartManoj** [comment]
> Also setting `LITELLM_DROP_PARAMS` environment is enough. [BerriAI/litellm@`3a35a58`/litellm/__init__.py#L76](https://github.com/BerriAI/litellm/blob/3a35a58859a145a4a568548316a1930340e7440a/litellm/__init__.py#L76)

https://github.com/BerriAI/litellm/issues/4175

**Comment by SmartManoj** [comment]
> > > Also setting `LITELLM_DROP_PARAMS` environment is enough. [BerriAI/litellm@`3a35a58`/litellm/**init**.py#L76](https://github.com/BerriAI/litellm/blob/3a35a58859a145a4a568548316a1930340e7440a/litellm/__init__.py#L76)
> > 
> > 
> > [BerriAI/litellm#4175](https://github.com/BerriAI/litellm/issues/4175)
> 
> Lets wait till the users from those 2 issues had time to test the LITELLM_DROP_PARAMS then.

I did. It worked.

**Comment by isavita** [comment]
@tobitege also comment out 
```python
            stop=[
                '</execute_ipython>',
                '</execute_bash>',
                '</execute_browse>',
            ],
```
in
```python
response = self.llm.do_completion(
            messages=messages,
            stop=[
                '</execute_ipython>',
                '</execute_bash>',
                '</execute_browse>',
            ],
            temperature=0.0,
        )
```
would work as well.

**Comment by tobitege** [comment]
It's not my code, so I'd be reaaally careful, to not just remove anything LLM-related there. 😁
And custom Stop params work for many other providers, so they have a purpose.

**Comment by isavita** [comment]
@tobitege I removed it on my branch, and of course, it works fine. However, I am not proposing this change for merging with the main code. So far all mistral models are too week to be useful.

**Comment by SmartManoj** [comment]
> so they have a purpose.

It will give only one action. Else, the prompt will be polluted.

File: opendevin/llm/llm.py
Comment (SmartManoj) [inline opendevin/llm/llm.py:96]:
```suggestion
            litellm.drop_params = self.config.drop_params
```

Comment (tobitege) [reply opendevin/llm/llm.py]: Thanks!

File: opendevin/core/schema/config.py
Comment (SmartManoj) [inline opendevin/core/schema/config.py:7]: Why is this config needed in the frontend?

Comment (enyst) [reply opendevin/core/schema/config.py:7]: This enum has been holding all or almost all settings, because we expect to implement them in FE at some point. Almost none of these was needed when added, they were added for future use.

File: opendevin/llm/llm.py
Comment (enyst) [inline opendevin/llm/llm.py:96]: This should work. It seems also possible that litellm accepts it in the completion call 🤔, according to their latest code. (the two links to litellm in a comment above have had some changes two weeks ago and they are sending it in `completion` at least to openai/azure)

**Review by enyst — APPROVED** [review]
@tobitege LGTM! I didn't test this, but if we can have it tested it's good to go. It makes sense to offer the ability to set this in config as usual. Thanks!

**Comment by tobitege** [comment]
> @tobitege LGTM! I didn't test this, but if we can have it tested it's good to go. It makes sense to offer the ability to set this in config as usual. Thanks!

Thanks!
Let me re-test this before merging so it doesn't break anything.

**Pull request merged**
